src/tls/certificate.c

```diff
@@ -40,8 +40,11 @@ static Certificate *
 certificate_new (gnutls_certificate_credentials_t creds)
 {
   Certificate *self = malloc (sizeof (Certificate));
-  self->creds = creds;
-  self->ref_count = 1;
+  if (self)
+    {
```

Comment:
Let's rather use mallocx() here (from src/common/cockpitmemory.h). Returning a `NULL` doesn't really help either, it's just going to crash later on.

Reply:
Done